bundles/org.openhab.voice.mactts/src/main/java/org/openhab/voice/mactts/internal/MacTTSAudioStream.java

```diff
@@ -80,12 +84,19 @@ public AudioFormat getFormat() {
     private InputStream createInputStream() throws AudioException {
         String outputFile = generateOutputFilename();
         String command = getCommand(outputFile);
+        logger.debug("Executing on command line: {}", command);
 
         try {
             Process process = Runtime.getRuntime().exec(command);
             process.waitFor();
             file = new File(outputFile);
+            if (!file.exists()) {
+                throw new AudioException("Generated file '" + outputFile + "' does not exist.'");
+            }
             this.length = file.length();
+            if (this.length == 0) {
+                throw new AudioException("Generated file '" + outputFile + "' has no content.'");
+            }
```

Comment:
Shouldn't these be check _before_ executing the command?

Reply:
No, because the command creates the file :-)